File: src/main/java/k31/grc/ast/node/cond/CondAnd.java
package k31.grc.ast.node.cond;

import k31.grc.ast.visitor.Visitor;

public class CondAnd extends CondBase {

	private CondBase left;
	private CondBase right;

	public CondBase getLeft() {
		return left;
	}

	public CondBase getRight() {
		return right;
	}

	public void setLeft(CondBase left) {
		this.left = left;
	}

	public void setRight(CondBase right) {
		this.right = right;
	}

	public CondAnd(String text) {
		super(text);
	}

	@Override
	public void accept(Visitor v) {

		v.visit(this);
	}
}

File: DotNetGrc/Grc/Ast/Node/Cond/CondAnd.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Grc.Ast.Visitor;

namespace Grc.Ast.Node.Cond
{
	public partial class CondAnd : CondBase
	{
		private CondBase left;
		private CondBase right;

		private string operAnd;

		public CondBase Left { get { return left; } }

		public CondBase Right { get { return right; } }

		public override int Line { get { return left.Line; } }

		public override int Pos { get { return left.Pos; } }

		public CondAnd(CondBase left, CondBase right, string operAnd)
		{
			this.left = left;
			this.right = right;

			this.operAnd = operAnd;
		}

		public override void Accept(IVisitor v)
		{
			v.Visit(this);
		}

		protected override string GetText()
		{
			return string.Format("({0} {1} {2})", left, operAnd, right);
		}

		public override string ToString()
		{
			return operAnd;
		}
	}
}
